exoctk/phase_constraint_overlap/phase_constraint_overlap.py

```diff
@@ -51,81 +76,448 @@ def calculate_phase(period, obsDur, winSize):
+            minphase = 1. + minphase
+        if maxphase > 1:
+            maxphase = maxphase - 1.
 
     return minphase, maxphase
 
-def calculate_obsDur(transitDur):
-    ''' Function to calculate the min and max phase.
+def calculate_preDur(transitDur):
+    ''' Function to calculate the pre-transit hours to be spent on target as recommended by the
+        Tdwell equation:
+        
+            0.75 + Max(1hr,T14/2) (before transit) + T14 + Max(1hr, T14/2) (after transit) + 1hr (timing window)
+
+        The output is, thus, 0.75 + Max(1hr,T14/2) (before transit) + T14/2.
 
         Parameters
         ----------
         transitDur : float
-            The duration of the transit in hours.
+            The duration of the transit/eclipse in hours.
+
+        Returns
+        -------
+        pretransit_duration : float
+            The duration of the observation prior to transit/eclipse mid-time in hours. '''
+
+    pretransit_duration = 0.75 + np.max([1., transitDur/2.]) + transitDur/2.
+
+    return pretransit_duration
+
+def drsky_2prime(x, ecc, omega, inc):
+    ''' Second derivative of function drsky
+
+    This is the second derivative with respect to f of the drsky function. 
+
+    Parameters
+    ----------
+
+    x : float
+      True anomaly
+    ecc : float
+      Eccentricity of the orbit
+    omega : float
+      Argument of periastron passage (in radians)
+    inc : float
+      Inclination of the orbit (in radians) 
+      
+    Returns
+    -------
+    drsky_2prime : float
+      Function evaluated at x, ecc, omega, inc'''
+
+    sq_sini = np.sin(inc)**2
+    sin_o_p_f = np.sin(x+omega)
+    cos_o_p_f = np.cos(x+omega)
+    ecosf = ecc*np.cos(x)
+    esinf = ecc*np.sin(x)
+
+    f1 = esinf - esinf*sq_sini*(sin_o_p_f**2)
+    f2 = -sq_sini*(ecosf + 4.)*(sin_o_p_f*cos_o_p_f)
+
+    return f1+f2
+
+def drsky_prime(x, ecc, omega, inc):
+    ''' Derivative of function drsky
+
+    This is the first derivative with respect to f of the drsky function. 
+
+    Parameters
+    ----------
+
+    x : float
+      True anomaly
+    ecc : float
+      Eccentricity of the orbit
+    omega : float
+      Argument of periastron passage (in radians)
+    inc : float
+      Inclination of the orbit (in radians) 
+      
+    Returns
+    -------
+    drsky_prime : float
+      Function evaluated at x, ecc, omega, inc'''
+
+    sq_sini = np.sin(inc)**2
+    sin_o_p_f = np.sin(x+omega)
+    cos_o_p_f = np.cos(x+omega)
+    ecosf = ecc*np.cos(x)
+    esinf = ecc*np.sin(x)
+
+    f1 = (cos_o_p_f**2 - sin_o_p_f**2)*(sq_sini)*(1. + ecosf)
+    f2 = -ecosf*(1 - (sin_o_p_f**2)*(sq_sini))
+    f3 = esinf*sin_o_p_f*cos_o_p_f*sq_sini
+    
+    return f1+f2+f3
+
+def drsky(x, ecc, omega, inc):
+    ''' Function whose roots we wish to find to obtain time of secondary (and primary) eclipse(s)
+
+    When one takes the derivative of equation (5) in Winn (2010; https://arxiv.org/abs/1001.2010v5), and equates that to zero (to find the 
```

Comment:
@nespinoza PEP257 suggests to limit lines of docstrings to 72 characters.

Reply:
I promise to get better at this in the future, but I think we really need this explaination here. Otherwise, myself (and future astronomers) won't know what this is about.